main.nf

```diff
@@ -988,7 +988,11 @@ workflow.onError {
                 break;
             }
         }
-        species_name = file("${configfile}").readLines().find{ it =~ /focal_species/ }.split()[2].strip()
+
+        focal_species_line = file("${configfile}").readLines().find{ it =~ /focal_species/ }
+        if ( focal_species_line.split().size() == 3 ) {
+            species_name = focal_species_line.split()[2].strip()
+        }
```

Comment:
This should have an `else`. What if `.size()` != 3? Then `species_name` is probably `null` and the following code will not to work correctly, e.g. the `log_filename` will probably be `rate_adjustment/null/logs_...` and the existing log file will not be found.

Also, does `.split()` and `.size()`/`.strip()` work correctly if someone has `focal_species=...` instead of `focal_species = ...`?

Reply:
Right, with `focal_species= elaeis` was not working. I've changed this slightly by making use of a `split("=")`.

Changed this also in `workflow.onComplete` section.

About the `else`, right now the code indeed doesn't find any log filename, what would you put in the `else` clause?

Reply:
Either define `species_name` in a way that makes the following code work correctly (`""` might work?), or execute appropriate other code.

Reply:
I think that `""` is not much better than `null`, the `log_filename` won't be a valid path with any of the two.
(I pushed the changes with `split("=")`, yesterday I only committed them...)

Reply:
I think `rate_adjustment/logs_` should be the correct path if `focal_species = ""`?

Otherwise you need to handle it separately somehow, like log a message that `focal_species` is not defined and one you should look in `rate_adjustment/` for log files or something like that...

Reply:
I'm now going with the second solution you proposed: logging a message in the error box about the focal species not being defined and pointing to the log folder without using the full path:
![image](https://user-images.githubusercontent.com/57489957/130220397-2daee7fc-a7ef-4c9a-a5e2-46ad97ea5e59.png)

Reply:
OK, but the "The error message can be found under rate_adjustment in logs_8301f6ca/setup_adjustment.log" looks like exactly what you would get with a simple `species_name = ""` resulting in a `log_filename = rate_adjustment/logs_8301f6ca/setup_adjustment.log`, no?

Reply:
I don't like too much the double slash between rate_adjustment and the log folder.

However, actually the code does something else right now xD If "species" is not defined, main.nf makes the log folder directly in rate_adjustment. I didn't realize that, it shouldn't happen! I'll fix this in the main.nf code with a check in the main.nf
![image](https://user-images.githubusercontent.com/57489957/130228752-cb2d00c5-d5a2-4959-9a37-25d11da1065f.png)
![image](https://user-images.githubusercontent.com/57489957/130228855-5c4d8b27-9aab-43de-bcfb-00d373373015.png)

Reply:
The double slash is a very easy fix. But either way, then it finds the log file correctly.

Reply:
Maybe it's not a bad idea to make the log folder directly into rate_adjustment... Without knowing the focal species name is otherwise not possible to generate the log file into a rate_adjustment/species folder. So we say to look into `rate_adjustment/logs_8301f6ca/setup_adjustment.log` _with only one slash_.

Reply:
I just noticed that `rate_adjustment/logs_39d729d7/` and `rate_adjustment**//**logs_39d729d7/` are both valid paths in the terminal... I thought not, then we can also leave the double slash.

Reply:
> Maybe it's not a bad idea to make the log folder directly into rate_adjustment... Without knowing the focal species name is otherwise not possible to generate the log file into a rate_adjustment/species folder.

Yes (unless you terminate everything right there at the beginning of `setupAdjustment` without generating any folders and log files).

> So we say to look into rate_adjustment/logs_8301f6ca/setup_adjustment.log with only one slash.

Which you can get with a simple `species_name = ""` in the `else` (and a small change in constructing `log_filename`) ;)

Reply:
> I just noticed that `rate_adjustment/logs_39d729d7/` and `rate_adjustment//logs_39d729d7/` are both valid paths in the terminal... I thought not, then we can also leave the double slash.

Yes, they are :) I thought you didn't like the look of it (and I would agree) ;)

Reply:
I'll close this: we decided to use `species=""` and to leave the `//`.
If you want feel free to change to single slash of course.